### ISSUE TITLE
vimdot not displaying graph on darwin

### ISSUE THREAD

**Issue by henningphan** [NixOS/nixpkgs#158764]
### Describe the bug
When using vimdot only vim starts, there is no graphical viewer of the graph

### Steps To Reproduce
1. Get a mac monterey
2. nix-env -iA nixpkgs.graphviz
3. vimdot

### Expected behavior
When performing the third step I expect two things, firstly vim to start, and secondly a gui window to show me the graph

### Screenshots
![vimdot](https://user-images.githubusercontent.com/1679039/153164402-9cab4409-de15-4e8d-aba0-9c6f64b60f2f.png)

### Additional context
```
henningphan@mackan:~$ dot -Txlib noname.gv
Format: "xlib" not recognized. Use one of: bmp canon cmap cmapx cmapx_np dot dot_json eps fig gd gd2 gif gv imap imap_np ismap jpe jpeg jpg json json0 mp pdf pic plain plain-ext png pov ps ps2 svg svgz tga tif tiff tk vdx vml vmlz vrml wbmp webp xdot xdot1.2 xdot1.4 xdot_json
henningphan@mackan:~$
```

### Notify maintainers
@bjornfor @7c6f434c 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
these 2 paths will be fetched (0.01 MiB download, 0.14 MiB unpacked):
  /nix/store/1vf7mgrsymraipkqf0pf1m0629lzxdkc-nix-info
  /nix/store/rqv2pkd4pj3c5r0154bl066j55g2xj6z-DarwinTools-1
copying path '/nix/store/rqv2pkd4pj3c5r0154bl066j55g2xj6z-DarwinTools-1' from 'https://cache.nixos.org'...
copying path '/nix/store/1vf7mgrsymraipkqf0pf1m0629lzxdkc-nix-info' from 'https://cache.nixos.org'...
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.3.0, macOS 12.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.6.0`
 - channels(henningphan): `""`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

```

**Comment by 7c6f434c** [comment]
My skimming of the format list makes me doubt that the current Darwin build supports any way of drawing that can create a window on macOS. Apparently the intent is to build X11 backend on Darwin, too, but something fails; maybe due to upstream changes. If there is ever a pull request that improves this on macOS while being an almost-no-op on Linux, I will be fine merging it. Maybe Nixpkgs Darwin maintenance team members know something about standard tricks on macOS, I myself have no idea.

**Comment by henningphan** [comment]
Any item in the format list that provides that live graph like xlib or any way that can easily provide me such feature?

**Comment by 7c6f434c** [comment]
No idea. You can try going through the list skipping the obvious file formats? Maybe Tk creates some kind of a window (maybe it needs something of Tcl/Tk in $PATH)

**Comment by henningphan** [comment]
Well I think I got what I needed, what should I do with the ticket? Leave it open as the functionality on darwin is limited?

**Comment by henningphan** [comment]
I tried some dot invocation and it wasnt really helpful, the best I could think of would be using e.g. using dot to produce jpg and some webserver to see it on the browser which automatically refresh it by using inotify.

**Comment by 7c6f434c** [comment]
Yeah makes sense. I guess whether to close or to leave open this issue won't change much the time until someone on macOS gets around to tuning the build to have any kind of updatable windowed output.

**Comment by realshovanshah** [comment]
I came across the same issue while trying to play with `graphviz`.
